Client/Frontend/AuthenticationManager/AuthenticationSettingsViewController.swift

```diff
@@ -124,7 +124,19 @@ class TouchIDSetting: Setting {
     private weak var navigationController: UINavigationController?
     private weak var switchControl: UISwitch?
 
-    init(title: NSAttributedString?, navigationController: UINavigationController? = nil, delegate: SettingsDelegate? = nil, enabled: Bool? = nil) {
+    private var touchIDSuccess: (() -> Void)? = nil
+    private var touchIDFallback: (() -> Void)? = nil
+
+    init(
+        title: NSAttributedString?,
+        navigationController: UINavigationController? = nil,
+        delegate: SettingsDelegate? = nil,
+        enabled: Bool? = nil,
+        touchIDSuccess: (() -> Void)? = nil,
+        touchIDFallback: (() -> Void)? = nil)
```

Comment:
I can't actually make these weak since they are non-class types. I think the retain cycle occurs because the methods passed in are implicitly bind to self since that's where they are defined. I think the solution is to pass in a closure with [unowned] self and call the method instead of passing it in directly.